src/components/AddBlockButton/AddBlockButton.tsx

```diff
@@ -14,23 +14,51 @@ export enum AddBlockButtonDirection {
     Vertical = 'Vertical',
 }
 
+export enum ButtonDirection {
+    Left = 'Left',
+    Top = 'Top',
+    Bottom = 'Bottom',
+    Right = 'Right',
+}
+
 export type AddBlockButtonProps = {
     onClick: () => void;
     title?: string;
+    /** @deprecated use direction with ButtonDirection */
     orientation?: AddBlockButtonDirection;
+    direction?: ButtonDirection;
     'data-test-id'?: string;
 };
 
 export const AddBlockButton = ({
     onClick,
     title,
-    orientation = AddBlockButtonDirection.Horizontal,
+    direction,
+    orientation,
     'data-test-id': dataTestId = 'add-block-button',
 }: AddBlockButtonProps): ReactElement => {
     const { isFocusVisible, focusProps } = useFocusRing();
     const ref = useRef<HTMLButtonElement | null>(null);
     const { buttonProps } = useButton({ onPress: () => onClick() }, ref);
 
+    let rotatePosition;
```

Comment:
This seems to be a bit overkill. Too much if condition here and also on line 70.
We can simplify this by using just the right conditions and mapping through the type using Record.

Remove this whole block, we don't need it and outside the component after line 22, I would just add those:
```
const orientationMap: Record<AddBlockButtonDirection, string> = {
    [AddBlockButtonDirection.Horizontal]: 'tw-rotate-0',
    [AddBlockButtonDirection.Vertical]: 'tw-rotate-90',
};

const directionMap: Record<ButtonDirection, string> = {
    [ButtonDirection.Left]: 'tw-rotate-180',
    [ButtonDirection.Top]: 'tw-rotate-[270deg]',
    [ButtonDirection.Bottom]: 'tw-rotate-90',
    [ButtonDirection.Right]: 'tw-rotate-0',
};
```

Reply:
Added